tests/functional/api.php

```diff
@@ -0,0 +1,93 @@
+];
+
+$number = [
+    ['?action=coverage&locales[]=it&locales[]=fr', 200, '1.92'],
+    ['?action=coverage&locales[]=fr', 200, '0.32'],
+    ['?action=coverage&locales[]=foo', 200, '0.00'],
+];
+
+$obj = new \pchevrel\Verif('Check API HTTP responses');
+$obj->setHost('localhost:8083');
+
+$check = function ($object, $paths, $type) {
+        foreach ($paths as $values) {
+            list($path, $http_code, $content) = $values;
+            switch ($type) {
+                case 'json':
+                    $object
+                        ->setPath($path)
+                        ->fetchContent()
+                        ->hasResponseCode($http_code)
+                        ->isJson()
+                        ->isEqualTo($content);
+                break;
+                case 'number':
+                    $object
+                        ->setPath($path)
+                        ->fetchContent()
+                        ->hasResponseCode($http_code)
+                        ->isNumeric()
+                        ->isEqualTo($content);
+                break;
+                default:
+                    $object
+                        ->setPath($path)
+                        ->fetchContent()
+                        ->hasResponseCode($http_code)
+                        ->isEqualTo($content);
+                break;
+            }
+        }
+};
+
+$check($obj, $json, 'json');
+$check($obj, $text, 'text');
+$check($obj, $number, 'number');
+
+$obj->report();
+
+// Kill PHP dev server we launched in the background
+exec('pkill -P ' . $pid);
```

Comment:
Use ```kill -9 ``` or the process won't die, ```pkill -P``` is to kill the descendant processes (when we launch a shell that itself launches a php server for example), in this script, we launch the php process directly, not via a shell script.